src/goto-symex/build_goto_trace.cpp

```diff
@@ -286,6 +294,19 @@ void build_goto_trace(
   for(auto &t_it : time_map)
     goto_trace.steps.splice(goto_trace.steps.end(), t_it.second);
 
+  // cut off the trace at the desired end or after the first failed
```

Comment:
In test generation, wouldn't we cut off traces that cover several test goals at the first goal that is reached? We would like to see the full trace to the end of the SSA.